cas-server-support-rest/src/main/java/org/jasig/cas/support/rest/TicketsResource.java

```diff
@@ -111,20 +115,17 @@
     @RequestMapping(value = "/tickets/{tgtId:.+}", method = RequestMethod.DELETE)
     public final ResponseEntity<String> deleteTicketGrantingTicket(@PathVariable("tgtId") final String tgtId) {
         this.cas.destroyTicketGrantingTicket(tgtId);
-        return new ResponseEntity<String>(tgtId, HttpStatus.OK);
+        return new ResponseEntity<>(tgtId, HttpStatus.OK);
     }
 
     /**
-     * Obtain credential from the request. Could be overridden by subclasses.
-     *
-     * @param requestBody raw entity request body
-     * @return the credential instance
+     * Default implementation of CredentialFactory.
      */
-    protected Credential obtainCredential(final MultiValueMap<String, String> requestBody) {
-        return new UsernamePasswordCredential(requestBody.getFirst("username"), requestBody.getFirst("password"));
-    }
+    public static class DefaultCredentialFactory implements CredentialFactory {
 
-    public CentralAuthenticationService getCentralAuthenticationService() {
-        return this.cas;
+        @Override
+        public Credential fromRequestBody(@NotNull MultiValueMap<String, String> requestBody) {
+            return new UsernamePasswordCredential(requestBody.getFirst("username"), requestBody.getFirst("password"));
```

Comment:
I'll get this merged in for now. If you could do a separate PR for the change, that would be great.